web/html/src/manager/minion/details/support-data.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from "react";
+import { useState } from "react";
+
+import { SubmitButton } from "components/buttons";
+import { DateTime, Form, Select, Text } from "components/input";
+import { ActionLink } from "components/links";
+import { Messages, MessageType, Utils as MessagesUtils } from "components/messages/messages";
+import { TopPanel } from "components/panels";
+import Validation from "components/validation";
+
+import { localizedMoment } from "utils";
+import Network from "utils/network";
+
+export interface UploadRegion {
+  label: string;
+  description: string;
+}
+
+class SupportDataRequest {
+  public caseNumber: string;
+  public region: string;
+  public parameters: string;
+  public earliest: moment.Moment;
+
+  public constructor(caseNumber: string, region: string, parameters: string, earliest: moment.Moment) {
+    this.caseNumber = caseNumber;
+    this.region = region;
+    this.parameters = parameters;
+    this.earliest = earliest;
+  }
+}
+
+interface Props {
+  serverId: number;
+  availableRegions: UploadRegion[];
+  supportProgramName: string | null;
+}
+
+export const SupportData: React.FC<Props> = ({ serverId, availableRegions, supportProgramName }): JSX.Element => {
+  const [formModel, setFormModel] = useState(
+    () => new SupportDataRequest("", availableRegions[0].label, "", localizedMoment())
```

Comment:
Will there always be at least one element in `availableRegions`?

Reply:
Yes, there will always be at least one location for the upload. I introduced a type `UploadRegionArray` to enforce that through typescript, even though as we are passing the values in the jade template, it is mostly for documenting it at the moment.